File: front/scada-front/src/app/DTOs/AnalogOutputDTO.ts
export interface AnalogOutputDTO {
    Name: string;
    Description: string;
    Address: string;
    InitialValue: number;
    LowLimit: number;
    HighLimit: number;
    Unit: string
  }

export function createAnalogOutputDTO(Name: string, Description: string, Address: string, InitialValue: number, LowLimit: number, HighLimit: number, Unit: string): AnalogOutputDTO {
    return {Name, Description, Address, InitialValue, LowLimit, HighLimit, Unit};
  }

export interface AnalogOutputIdDTO {
    Id: number;
    Name: string;
    Description: string;
    Address: string;
    InitialValue: number;
    LowLimit: number;
    HighLimit: number;
    Unit: string
  }

export function createAnalogOutputIdDTO(Id: number, Name: string, Description: string, Address: string, InitialValue: number, LowLimit: number, HighLimit: number, Unit: string): AnalogOutputIdDTO {
    return {Id, Name, Description, Address, InitialValue, LowLimit, HighLimit, Unit};
  }

File: back/scada-back/DTOs/AnalogOutputDTO.cs
namespace scada_back.DTOs
{
    public class AnalogOutputDTO
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public string Address { get; set; }

        public double InitialValue { get; set; }

        public double LowLimit { get; set; }

        public double HighLimit { get; set; }

        public string Unit { get; set; }
    }
}
